docs/sequence-diagrams/rule.puml

```diff
@@ -0,0 +1,3 @@
+@startuml rule
+thanos_rule -> thanos_store: data
```

Comment:
This is not helpful, and quite misleading ): There is many things happening here. Let's drop this for now and introduce in another PRs with details what do you think?